game/src/info/trip.rs

```diff
@@ -386,6 +389,329 @@ pub fn cancelled(ctx: &mut EventCtx, app: &App, id: TripID) -> Widget {
+                                display_txt.add(Line(format!("  Average speed: {}", avg_speed)));
+                                segments.push((lane_detail.length(), Color::RED, display_txt));
+                                norm_distance = Distance::meters(0.0);
+                                found = true;
+                                break;
+                            }
+                        }
+                        if !found {
+                            norm_distance += lane_detail.length();
+                        }
+                    }
+                    PathStep::Turn(id) => {
+                        let turn_details = map.get_t(*id);
+                        sum_phase_dist += turn_details.geom.length();
+                        let mut found = false;
+                        for (turn, delay) in &data_inter {
+                            if turn == id {
+                                segments.push((
+                                    norm_distance,
+                                    norm_color,
+                                    build_text(&msgs, &norm_distance),
+                                ));
+
+                                let mut display_txt = Text::from(Line(&p.phase_type.describe(map)));
+                                display_txt.add(Line(format!("  Intersection: {}", turn.parent)));
+                                display_txt.add(Line(format!(
+                                    "  Delay: {}",
+                                    Duration::seconds(*delay as f64)
+                                )));
+                                segments.push((
+                                    // To make sure that the hotspot isn't too small
+                                    if 0.05 < (turn_details.geom.length() / sum_phase_dist) {
+                                        turn_details.geom.length()
+                                    } else {
+                                        sum_phase_dist += sum_phase_dist * 0.05;
+                                        sum_phase_dist -= turn_details.geom.length();
+                                        sum_phase_dist * 0.05
+                                    },
+                                    Color::RED,
+                                    display_txt,
+                                ));
+                                norm_distance = Distance::meters(0.0);
+                                found = true;
+                                break;
+                            }
+                        }
+                        if !found {
+                            norm_distance += turn_details.geom.length();
+                        }
+                    }
+                }
+            }
+            segments.push((norm_distance, norm_color, build_text(&msgs, &norm_distance)));
+        } else {
+            // TODO Think of something to do instead
+            unimplemented!("We dont have a path!!");
```

Comment:
I think this case actually happens sometimes (https://github.com/dabreegster/abstreet/issues/361), so instead of crashing, how about doing nothing or maybe `error!("No path for {}", trip_id)`

Reply:
I agree with the not crashing, it just means that the time line bar won't be made, or there may be issues with the bar. But I haven't encountered that scenario, so I haven't been able to test it yet.
I'll change it to the `error!` and leave the todo in, if that's okay?
Or maybe we could show a message to the user, (if the error doesn't already do that) saying that the timeline broke?

Reply:
An error! and TODO sound good for now. I'll see if I can get an easy way to reproduce a missing path, and see how the missing timeline looks (or just fix the missing path problem)